doc/proposals/images-subspec.md

```diff
@@ -0,0 +1,257 @@
+---
+title: Images Sub-specification
+authors:
+  - '@zachncst'
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-11-04
+last-updated: 2019-11-05
+status: provisional
+help-from:
+  - '@camilamacedo86'
+  - '@kevinrizza'
+  - '@shawn-hurley'
+see-also:
+  - 'https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/contributors/design-proposals/related-images.md'
+---
+
+# Images Sub Spec
+
+## Release Signoff Checklist
+
+- \[ \] Enhancement is `implementable`
+- \[ \] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created in
+  [operator-sdk/doc][operator-sdk-doc]
+
+## Tag Line
+
+> Who likes hard coding image locations anyways?
+
+## Summary
+
+When an operator is installed, the images that the operator installs as operands in the
+form of pods, deployments or other kubernetes resources is not always clearly
+defined and editable by a user of an operator. This proposal would define
+an operator-sdk ImageSpec and ImagesListSpec object structs that operator developers
+could use to enhance the creation of operator CRDs or improve existing
+operators. The spec objects would let operator users override image locations
+for that custom resource.
+
+Additionally, the proposal would provide tooling to generate scaffold code for
+creating new operators. When creating a new operator, an operator developer
+would be able to pass flags that would create a base spec object that includes
+the ImagesListSpec.
+
+## Motivation
+
+This proposal is driven by a few motivations:
+
+- Makes operators more transparent with what is being installed on the cluster.
+- Formalizes a best practice of operator development.
+- The depencency between operator version and operator resource version is broken.
+- Allows for out of band updates of the underlying images.
+- Registry information for the images can easily be overridden to point at
+  private registries.
+- Provides some (potential) standardization to the CRD.
+
+## Goals
+
+- Provide an easy mechanism of defining and overriding images and related image
+  information (like pull secrets).
+- Increase transparency with what operators are installing.
+- Simple, reusable specs to build operators; similar to PodSpec in Kubernetes.
+
+### Non-Goals
+
+- OLM management of the images spec is not in this proposal.
+- Mapping/re-mapping other container values (ports, etc.).
+- Additional pod information, like env vars.
+
+## Proposal
+
+Implementing the proposal would include:
+
+- Add an optional section to the CRD spec named images that contains definitions of
+  all images used by the operator. An example:
+
+  ```yaml
+  apiVersion: app.example.com/v1alpha1
```

Comment:
IMHO: 
- We should not create a YAML file now with the images and etc. 
- We could use a flag to inform the image and scaffold a common implementation instead of that. (Reasons: Give flexibility for changes, make transparent what the project is dong instead of having a block magic box that does things, we should began by doing this process for only 1 image in order to keep it simple and then, after we have it working and all improvements that will happen following an agile process we could see better it would be good or not allow it to be done for N/a list of images by just one action. Note that this approach will not just allow us to grow with consistency but make easier the process to implement it since if we have it working as expected for 1 image should not be a big effort to create a feature to allow N images. )
-